airflow/contrib/aws_glue_job_hook.py

```diff
@@ -0,0 +1,212 @@
+            RunId=run_id,
+            PredecessorsIncluded=True
+        )
+        job_run_state = job_status['JobRun']['JobRunState']
+        failed = job_run_state == 'FAILED'
+        stopped = job_run_state == 'STOPPED'
+        completed = job_run_state == 'SUCCEEDED'
+
+        while True:
+            if failed or stopped or completed:
+                self.log.info("Exiting Job {} Run State: {}"
+                              .format(run_id, job_run_state))
+                return {'JobRunState': job_run_state, 'JobRunId': run_id}
+            else:
+                self.log.info("Polling for AWS Glue Job {} current run state"
+                              .format(job_name))
+                time.sleep(6)
+
+    def get_or_create_glue_job(self):
+        glue_client = self.get_conn()
+        try:
+            self.log.info("Now creating and running AWS Glue Job")
+            s3_log_path = "s3://{bucket_name}/{logs_path}{job_name}"\
+                .format(bucket_name=self.s3_bucket,
+                        logs_path=self.S3_GLUE_LOGS,
+                        job_name=self.job_name)
+
+            execution_role = self.get_iam_execution_role()
+            script_location = self._check_script_location()
+            create_job_response = glue_client.create_job(
+                Name=self.job_name,
+                Description=self.desc,
+                LogUri=s3_log_path,
+                Role=execution_role['Role']['RoleName'],
+                ExecutionProperty={"MaxConcurrentRuns": self.concurrent_run_limit},
+                Command={"Name": "glueetl", "ScriptLocation": script_location},
+                MaxRetries=self.retry_limit,
+                AllocatedCapacity=self.num_of_dpus
+            )
+            # print(create_job_response)
+            return create_job_response
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to create aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def _check_script_location(self):
+        """
+        :return: S3 Script location path
+        """
+        if self.script_location[:5] == self.S3_PROTOCOL:
+            return self.script_location
+        elif os.path.isfile(self.script_location):
+            s3 = self.get_resource_type('s3', self.region_name)
+            script_name = os.path.basename(self.script_location)
+            s3.meta.client.upload_file(self.script_location,
+                                       self.s3_bucket,
+                                       self.S3_ARTIFACTS_PREFIX + script_name)
+
+            s3_script_path = "s3://{s3_bucket}/{prefix}{job_name}/{script_name}" \
+                .format(s3_bucket=self.s3_bucket,
+                        prefix=self.S3_ARTIFACTS_PREFIX,
+                        job_name=self.job_name,
+                        script_name=script_name)
+            return s3_script_path
```

Comment:
If it's uploaded to `self.script_location`, maybe update that instance variable to use the s3 location for it?  This private method might update a private instance variable to memoize this operation so it's not repeated unnecessarily.